eZ/Publish/API/Repository/Tests/SearchServiceFieldFiltersTest.php

```diff
@@ -29,14 +29,34 @@ public function setUp()
     {
         $setupFactory = $this->getSetupFactory();
 
-        if ( !$setupFactory instanceof LegacyElasticsearch )
+        if ( $setupFactory instanceof LegacySolr )
         {
-            $this->markTestIncomplete( "ATM implemented only for Elasticsearch storage" );
+            $this->markTestIncomplete( "Not implemented for Solr Search Engine" );
         }
 
         parent::setUp();
     }
 
+    protected function supportsFilteringFullText()
+    {
+        if ( ltrim( get_class( $this->getSetupFactory() ), '\\' ) === 'eZ\\Publish\\API\\Repository\\Tests\\SetupFactory\\Legacy' )
+        {
+            $this->markTestSkipped(
+                "Legacy Search Engine does not support field filters with Fulltext criterion"
+            );
+        }
+    }
+
+    protected function supportsCustomFields()
```

Comment:
I'd rename this to checkCustomFieldSupport(). "supports" kind of indicates that it will return a boolean, not skip the test if it doesn't.